lib/daru/maths/statistics/vector.rb

```diff
@@ -169,6 +169,30 @@ def variance_population m=nil
           end
         end
 
+        # Sample covariance with denominator (N-1)
+        def covariance_sample other
+          @size == other.size or raise ArgumentError, "size of both the vectors must be equal"
+          mean_x = self.mean 
+          mean_y = other.mean
+          sum = 0
+          (0...size).each do |i| 
+            sum = sum + ((@data[i].nil? || other.data[i].nil?) ? 0 : (@data[i] - mean_x) * (other.data[i] - mean_y))
```

Comment:
Missing data need only consist of `nil` elements. It can be anything in `missing_values`, even numbers.